services/api-server/src/simcore_service_api_server/modules/catalog.py

```diff
@@ -34,15 +43,124 @@ async def on_shutdown() -> None:
     app.add_event_handler("shutdown", on_shutdown)
 
 
+SolverNameVersionPair = Tuple[SolverName, str]
+
+
+class TruncatedServiceOut(ServiceDockerData):
+    """This is a partial replica of catalog's API response body schema
+    in services/catalog/src/simcore_service_catalog/models/schemas/services.py::ServiceOut
+
+    It used here to parse and extract only necessary information from the
+    response. Ideally the rest of the response is dropped so here it would
+    perhaps make more sense to use something like graphql
+    that asks only what is needed.
```

Comment:
I am not sure I understand that comment here. I mean you are inheritating from ServiceDockerData, so you get everything from it. well it's just a bit confusing. Basically you are only interested in the ServiceDockerData part.

Reply:
It is the other way around. I cannot use directly ``ServiceDockerData`` to partially parse the response of the catalog ``ServiceOut`` because the former declares
```python
    class Config:
        extra = Extra.forbid # < ----
```
So the solution is to inherit and override this restriction